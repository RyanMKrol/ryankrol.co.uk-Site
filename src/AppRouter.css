#root-container {
  height: 100%;
  width: 100%;
  position: absolute;
  background-color: #CCC;
}

.page-body {
  width: 90%;
  height: 100%;
  margin: 0 auto;
  background-color: white;
}
