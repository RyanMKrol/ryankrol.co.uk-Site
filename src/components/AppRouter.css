#root-container {
  display: flex;
  min-height: 100vh;
  flex-direction: column;
  background-color: #CCC;
}

.page-body {
  width: 90%;
  margin: 0 auto;
  background-color: white;
}
