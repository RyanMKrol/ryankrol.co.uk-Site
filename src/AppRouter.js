import React from "react"
import {
  BrowserRouter as Router,
  Switch,
  Route,
} from "react-router-dom"

import Home from "./pages/Home"
import About from "./pages/About"
import Header from "./components/Header"

import './AppRouter.css';

export default function AppRouter() {
  return (
    <div id="root-container">
      <Router>
        <Header />
        <div className="page-body">
          <Switch>
            <Route exact path="/">
              <Home />
            </Route>
            <Route path="/about">
              <About />
            </Route>
          </Switch>
        </div>
      </Router>
    </div>
  )
}
