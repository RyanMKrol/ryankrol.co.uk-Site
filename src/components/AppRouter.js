import React from "react"
import {
  BrowserRouter as Router,
  Switch,
  Route,
} from "react-router-dom"

import {
  Home,
  About,
  Portfolio,
  Music,
} from "./../pages"

import Header from "./Header"
import Footer from "./Footer"

import './AppRouter.css';

export default function AppRouter() {
  return (
    <div id="root-container">
      <Router>
        <Header />
        <div className="page-body">
          <Switch>
            <Route exact path="/">
              <Home />
            </Route>
            <Route path="/about">
              <About />
            </Route>
            <Route path="/portfolio">
              <Portfolio />
            </Route>
            <Route path="/music">
              <Music />
            </Route>
          </Switch>
        </div>
        <Footer />
      </Router>
    </div>
  )
}
