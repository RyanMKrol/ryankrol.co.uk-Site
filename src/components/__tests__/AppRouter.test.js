import React from 'react'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'

import AppRouter from './../AppRouter'

// Render Tests

it('renders router only when rendered', () => {
    const appRouter = shallow(<AppRouter />)
    expect(toJson(appRouter)).toMatchSnapshot()
})

it('does not crash when rendered', () => {
    expect(() => mount(<AppRouter />)).not.toThrow()
})
