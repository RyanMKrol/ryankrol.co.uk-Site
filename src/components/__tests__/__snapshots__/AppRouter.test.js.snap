// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders router only when rendered 1`] = `
<div
  id="root-container"
>
  <BrowserRouter>
    <withRouter(Header) />
    <div
      className="page-body"
    >
      <Switch>
        <Route
          exact={true}
          path="/"
        >
          <Home />
        </Route>
        <Route
          path="/about"
        >
          <About />
        </Route>
        <Route
          path="/portfolio"
        >
          <Portfolio />
        </Route>
      </Switch>
    </div>
    <Footer />
  </BrowserRouter>
</div>
`;
